File: software/examples/ruby/example_interrupt.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_io4'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change XYZ to the UID of your IO-4 Bricklet

ipcon = IPConnection.new # Create IP connection
io = BrickletIO4.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Register interrupt callback
io.register_callback(BrickletIO4::CALLBACK_INTERRUPT) do |interrupt_mask, value_mask|
  puts "Interrupt Mask: #{'%04b' % interrupt_mask}"
  puts "Value Mask: #{'%04b' % value_mask}"
  puts ''
end

# Enable interrupt on pin 0
io.set_interrupt 1 << 0

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect

File: software/examples/python/example_interrupt.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

HOST = "localhost"
PORT = 4223
UID = "XYZ" # Change XYZ to the UID of your IO-4 Bricklet

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_io4 import BrickletIO4

# Callback function for interrupt callback
def cb_interrupt(interrupt_mask, value_mask):
    print("Interrupt Mask: " + format(interrupt_mask, "04b"))
    print("Value Mask: " + format(value_mask, "04b"))
    print("")

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    io = BrickletIO4(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected

    # Register interrupt callback to function cb_interrupt
    io.register_callback(io.CALLBACK_INTERRUPT, cb_interrupt)

    # Enable interrupt on pin 0
    io.set_interrupt(1 << 0)

    raw_input("Press key to exit\n") # Use input() in Python 3
    ipcon.disconnect()
